README.md

```diff
@@ -517,6 +517,14 @@ You first need to create an [Atlas account](https://atlas.hashicorp.com/account/
 #### Options:
 
 * **github-token**: GitHub oauth token with `repo` permission.
+* **deploy-key**: Filename of an encrypted
+  [GitHub deploy key](https://developer.github.com/v3/guides/managing-deploy-keys/#deploy-keys)
+  with write access to the **repo** repository, encrypted with `travis
+  encrypt-file`.
+* **deploy-key-key**: The key produced by `travis encrypt-file`, usually of the
```

Comment:
Why is this necessary?

This discourages use with the repositories which already has encrypted file. https://docs.travis-ci.com/user/encrypting-files#Encrypting-multiple-files It is cumbersome, if not impossible, to add this feature.

Reply:
Oh geez, that's awful. Why wouldn't it just add a new $encrypted_* pair? 

Anyway, it's clearly necessary to encrypt the deploy key: otherwise anyone with read access to the repository could also write to it. I can see a couple ways to avoid travis-ci/travis.rb#239:

1. Point people to https://docs.travis-ci.com/user/encrypting-files#Using-OpenSSL to generate the encrypted deploy-key file, and provide two parameters, one pointing at the file, and the other to the decryption key. I like this option best, so I'll start on it.
1. Have folks decrypt the deploy key in their `before_deploy` steps, and just point to it here. This has the two downsides that 1) it's more code to write in `.travis.yml`, and 2) it makes it easy to use an insecure deploy key checked directly into the repository.
1. Have two variables to find the deploy key file: 1 identifying the encrypted tarfile, and the other pointing to the deploy key inside it.

Reply:
Done. New test build at https://travis-ci.org/jyasskin/testing/builds/290726967.

Reply:
It is still not clear on *why* the use of "deploy key" as a separate file is necessary. In other deployment providers, deployment keys are given as a bare string (whether or not it is encrypted). For example,

```yaml
deploy:
  provider: releases
  api_key:
    secure: YOUR_API_KEY_ENCRYPTED
⋮
```

It is not clear to me why this provider *requires* an encrypted file to deploy (or have that option at all). If the requirement is *having* the `#{gitdir}/deploykey` file, then it is easy to write that file from what is given in the configuration.